src/PowerShellEditorServices/Services/PowerShell/Debugging/PowerShellDebugContext.cs

```diff
@@ -83,22 +79,21 @@ public void StepOver()
 
         public void SetDebugResuming(DebuggerResumeAction debuggerResumeAction)
         {
-            _consoleRepl?.SetReplPop();
+            _psesHost.SetExit();
             LastStopEventArgs.ResumeAction = debuggerResumeAction;
-            _debugLoopCancellationSource.Cancel();
+            // We need to tell whatever is happening right now in the debug prompt to wrap up so we can continue
+            _psesHost.CancelCurrentTask();
         }
 
+        // This must be called AFTER the new PowerShell has been pushed
         public void EnterDebugLoop(CancellationToken loopCancellationToken)
         {
-            _debugLoopCancellationSource = CancellationTokenSource.CreateLinkedTokenSource(loopCancellationToken);
             RaiseDebuggerStoppedEvent();
-
         }
 
+        // This must be called BEFORE the debug PowerShell has been popped
```

Comment:
I thought about deleting this method, but it might turn out we need to put some state cleanup in here